draft-ietf-quic-multipath.md

```diff
@@ -1180,31 +1180,79 @@ but has not provided a corresponding connection ID for the path ID
 Connection IDs are separately allocated for each Path ID.
-Rotating the connection ID on a path does not change the Path ID.
-NAT rebinding, though it changes the 4-tuple of the path,
-also does not change the path identifier.
-The packet number space does not change when connection ID
-rotation happens within a given Path ID.
+Rotating the connection ID on a path does not change the Path ID
+nor packet number space.
+
+While endpoints assign a connection ID to a specific sending 4-tuple,
+networks events such as NAT rebinding may make the packet's receiver
+observe a different 4-tuple. Though it changes the 4-tuple of the path,
+also does not change connection ID and as such also not the path identifier.
+Servers observing a 4-tuple change will
+perform path validation (see {{Section 9 of QUIC-TRANSPORT}}).
+If path validation process succeeds, the endpoints set
+the path's congestion controller and round-trip time
+estimator according to {{Section 9.4 of QUIC-TRANSPORT}}.
+
+{{Section 9.3 of QUIC-TRANSPORT}} allows an endpoint to skip validation of
+a peer address if that address has been seen recently. However, when the
+multipath extension is used and an endpoint has multiple addresses that
+could lead to switching between different paths, it should rather maintain
+multiple open paths instead.
+
+More generally, while migration cannot be avoided in case of network-based
+NAT rebindings, opening a new path instead of active client migration
+should be strongly preferred when the multipath extension is supported.
+This enables a smoother handover and allows a simplified migration
+handling at the server as NAT rebindings imply immediate loss of the old
+address.
+
+## Using multiple paths on the same 4-tuple
+
+It is possible to create paths that
+refer to the same 4-tuple. For example, the endpoints may want
+to create paths that use different Differentiated Service {{?RFC2475}} markings.
+This could be done in conjunction with scheduling algorithms
+that match streams to paths, so that for example data frames for
+low priority streams are sent over low priority paths.
+Since these paths use different path IDs, they can be managed
+independently to suit the needs of the application.
+
+There may be cases in which paths are created with different 4-tuples,
+but end up using the same 4-tuples as a consequence of path
+migrations. For example:
 
-Data associated with the transmission and reception such RTT measurements,
-congestion control state, or loss recovery are maintained per packet number
-space and as such per Path ID.
+* Client starts path 1 from address 192.0.2.1 to server address 198.51.100.1
+* Client starts path 2 from address 192.0.2.2 to server address 198.51.100.1
+* both paths are used for a while.
+* Server sends packet from address 198.51.100.1 to client address 192.0.2.1, with CID indicating path=2.
+* Client receives packet, recognizes a path migration, update source address of path 2 to 192.0.2.1.
```

Comment:
While I agree with the provided example, it implicitly makes the assumption that both paths use the same source and destination ports; otherwise, it would not lead to the same 4-tuple.

Reply:
Yes, should we add that or is it okay to just not say it explicitly?

Reply:
I would say, rather be explicit than implicit. This is not blocking though.